test/suite/test_assert06.py

```diff
@@ -0,0 +1,403 @@
+            'commit_timestamp=' + timestamp_str(timestamp))
+        self.session.timestamp_transaction(
+            'durable_timestamp=' + timestamp_str(timestamp))
+
+    def test_timestamp_alter(self):
+        base = 'assert04'
+        uri = 'file:' + base
+        cfg_on = 'assert=(durable_timestamp=key_consistent)'
+        cfg_off = 'assert=(durable_timestamp=none)'
+        msg_ooo='/out of order/'
+        msg_usage='/used inconsistently/'
+
+        # Create the table without the key consistency checking turned on.
+        # Create a few items breaking the rules. Then alter the setting and
+        # verify the inconsistent usage is detected.
+        self.session.create(uri, 'key_format=S,value_format=S')
+        # Insert a data item at timestamp 2.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value2'
+        self.apply_timestamps(2)
+        self.session.commit_transaction()
+        c.close()
+
+        # Modify the data item at timestamp 1.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value1'
+        self.apply_timestamps(1)
+        self.session.commit_transaction()
+        c.close()
+
+        # Insert a non-timestamped item. Then modify with a timestamp. And
+        # again modify without a timestamp.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value_nots'
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value2'
+        self.apply_timestamps(2)
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_nots'] = 'value_nots2'
+        self.session.commit_transaction()
+        c.close()
+
+        # We must move the oldest timestamp forward in order to alter.
+        # Otherwise alter closing the file will fail with EBUSY.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(2))
+
+        # Now alter the setting and make sure we detect incorrect usage.
+        self.session.alter(uri, cfg_on)
+
+        # Detect decreasing timestamp.
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value5'
+        self.apply_timestamps(5)
+        self.session.commit_transaction()
+        c.close()
+
+        c = self.session.open_cursor(uri)
+        self.session.begin_transaction()
+        c['key_ts1'] = 'value4'
+        self.session.prepare_transaction(
+            'prepare_timestamp=' + timestamp_str(4))
+        self.session.timestamp_transaction(
+            'commit_timestamp=' + timestamp_str(4))
+        self.session.timestamp_transaction(
+            'durable_timestamp=' + timestamp_str(4))
```

Comment:
Either the above should call `self.apply_timestamps(4)` or it should have a comment about why that isn't appropriate here. I see no differences so I think it should just call the function.